### ISSUE TITLE
[Question] Can I use Collie to get shopping cart complementary products on an e-shop?

### ISSUE THREAD

**Issue by almirb** [ShopRunner/collie#38]
Hi!
In a scenario where we have a list of product purchases on an e-shop, we want to predict what could be the complementary itens the user may buy together (aka customers who bought this item also bought...).

Do you think Collie can help us with that task?
Thanks!

**Comment by nathancooperjones** [comment]
Hi @almirb. 

Sorry for the delayed answer in this, but this is a great question! 

Although Collie is specifically optimized for user-item recommendations (input a user, output a series of recommended items), we also have the ability to do item-item recommendations. This is about equivalent to Amazon's `People who bought this also bought this...`. At ShopRunner, for instance, we have seen that getting the item-item recommendations for a pair of pants will include complementary shirts, shoes, etc. - since these are all items usually purchased or viewed together. 

In Collie, it's easy to get these item-item recommendations using the `item_item_similarity` method included with all Collie models: 

https://github.com/ShopRunner/collie/blob/0963998ded3eb566fda863ad462ab5bec6bcd223/collie/model/base/base_pipeline.py#L680

I think this is great for a V1, but if you wanted a more dedicated model for complementary items, it might be worth considering building an entirely separate model. My old coworker at ShopRunner, @sugi-chan, wrote a great blog post about building up a model specifically to find complementary items that work together in a single order: https://medium.com/shoprunner/a-tricksy-look-at-outfitting-kitsune-2dabff17c690. Whereas some Collie items return item-item recommendations that are just similar to the seed item (something like input a couch, output other couches people bought, which is not really complementary products), this model is specifically trained to output complementary items _always_. I think something like that is a great V2. 

TL;DR: Yes, Collie can easily output complementary products, but performance may be worse than a completely dedicated complementary product model. Regardless, Collie's `item_item_similarity` is surprisingly powerful.
